src/object/object_primitive.c

```diff
@@ -17601,26 +17591,113 @@ mr_data_cmpdisk_json (void *mem1, void *mem2, TP_DOMAIN * domain, int do_coercio
       goto cleanup;
     }
 
-  res = (strcmp (first_json_body, second_json_body) == 0 ? DB_EQ : DB_UNK);
+  rc = db_json_get_json_from_str (first_json_body, doc1);
+  assert (rc == NO_ERROR && doc1 != NULL);
+  rc = db_json_get_json_from_str (second_json_body, doc2);
+  assert (rc == NO_ERROR && doc2 != NULL);
+
+  db_make_json (&json1, first_json_body, doc1, true);
+  db_make_json (&json2, second_json_body, doc2, true);
+
+  res = mr_cmpval_json (&json1, &json2, do_coercion, total_order, 0, 0);
 
 cleanup:
-  db_private_free (NULL, first_json_body);
-  db_private_free (NULL, second_json_body);
+  pr_clear_value (&json1);
```

Comment:
please notice that error handling of `pr_get_compressed_data_from_buffer` is bad.
At the moment of jumping, `json1` and `json2` might be uninited and `first_json_body` and `second_json_body` might be leaked.

please also notice `strc` is unused.